x-pack/test/api_integration/apis/metrics_ui/log_threshold_alert.ts

```diff
@@ -0,0 +1,236 @@
+            [
+              {
+                actionGroup: 'logs.threshold.fired',
+                context: {
+                  conditions: 'env does not equal dev',
+                  group: null,
+                  isRatio: false,
+                  matchingDocuments: 2,
+                  reason: '2 log entries in the last 5 mins. Alert when ≥ 1.',
+                },
+              },
+            ],
+          ]);
+        });
+      });
+
+      describe('with group by', () => {
+        it('should work', async () => {
+          const timestamp = new Date(DATES['alert-test-data'].gauge.max);
+          const alertFactory = sinon.fake();
+          const ruleParams = {
+            count: {
+              comparator: Comparator.GT_OR_EQ,
+              value: 1,
+            },
+            timeUnit: 'm' as TimeUnit,
+            timeSize: 5,
+            groupBy: ['env'],
+            criteria: [
+              {
+                field: 'env',
+                comparator: Comparator.NOT_EQ,
+                value: 'test',
+              },
+            ],
+          };
+          await executeAlert(
+            ruleParams,
+            '@timestamp',
+            'alerts-test-data',
+            {},
+            esClient,
+            alertFactory,
+            timestamp.valueOf()
+          );
+          expect(alertFactory.callCount).to.equal(2);
+          expect(alertFactory.getCall(0).args).to.eql([
+            'dev',
+            '2 log entries in the last 5 mins for dev. Alert when ≥ 1.',
+            2,
+            1,
+            [
+              {
+                actionGroup: 'logs.threshold.fired',
+                context: {
+                  conditions: 'env does not equal test',
+                  group: 'dev',
+                  isRatio: false,
+                  matchingDocuments: 2,
+                  reason: '2 log entries in the last 5 mins for dev. Alert when ≥ 1.',
+                },
+              },
+            ],
+          ]);
+        });
+      });
+    });
+
+    describe('executeRatioAlert', () => {
+      before(() => esArchiver.load('x-pack/test/functional/es_archives/infra/ten_thousand_plus'));
+      after(() => esArchiver.unload('x-pack/test/functional/es_archives/infra/ten_thousand_plus'));
+
+      describe('without group by', () => {
+        it('should work', async () => {
+          const timestamp = new Date(DATES.ten_thousand_plus.max);
+          const alertFactory = sinon.fake();
+          const ruleParams = {
+            count: {
+              comparator: Comparator.GT_OR_EQ,
+              value: 0.5,
+            },
+            timeUnit: 'm' as TimeUnit,
+            timeSize: 5,
+            criteria: [
+              [{ field: 'event.dataset', comparator: Comparator.EQ, value: 'nginx.error' }],
+              [{ field: 'event.dataset', comparator: Comparator.NOT_EQ, value: 'nginx.error' }],
+            ] as RatioCriteria,
+          };
+          await executeRatioAlert(
```

Comment:
Would it be advantageous to call `executeAlert` here instead of `executeRatioAlert` directly? That way we'd cover the ratio branching decision as well.

Reply:
The level up from here is the actual executor itself which require stubbing out a bunch of extra stuff including all the Rule Registry lifecycle stuff. My ultimate goal is to have these tests in place so I can experiment with moving the evaluations into Elasticsearch while maintaining the interfaces for `executeAlert` and `executeRatioAlert`. Ideally we should be able to make changes underneath these two methods and the results should remain the same while pushing all of the evaluations down to Elasticsearch.